### ISSUE TITLE
Change API Name to "Mojaloop API"

### ISSUE THREAD

**Issue by matthewbohan** [mojaloop/mojaloop-specification#15]
For consistency, simplicity, and to take advantage of branding, update the name of the API to "Mojaloop API."  Implement using a global search and replace for text.  There will also likely be some additional effort to replace in tables and diagrams.

**Comment by elnyry** [comment]
The API - The Open API for FSP Interoperability (FSPIOP API)

**Comment by ehenrka** [comment]
I'm not in favor of changing the name to Mojaloop API. The idea of the API was that it should be open for anyone who wants to enable financial interoperability between FSPs. Changing the name to Mojaloop more or less restricts the API to be for the Mojaloop switch instead of being open, as the name Mojaloop is tightly coupled with the open source switch implementation.

A name change would also limit competition in the market. If I was implementing a switch, why should i opt to use the Mojaloop API instead of an API that is meant to be open (the Open API for FSP Interoperability)? The current name does not in any way restrict the potential usage of the API, rather the contrary.

I don't understand why it would be simpler to have the name Mojaloop API. Mojaloop API says nothing about what the API is meant to do, which means that you need to explain the purpose instead. Changing the name would also mean less constistency within the actual API, as there are for example headers named after the current API name. The argument "for consistency, simplicity" is too simplistic without more information on how a name change would be beneficial, and whom it would be better for.

There is also a cost of rebranding this within platforms which already support the API. We have documentation and presentation materials which is not free to update. It will not just be a search and replace to support a name change.

**Comment by MichaelJBRichards** [comment]
While I appreciate your reservations, Henrik, it seems to me that they really divide into two separate considerations. The first is whether there should be a name change at all, and the second is (assuming that the answer to the first question is yes) whether the proposed name is the right one.

My feeling is that we do need a new name. For me this is demonstrated sufficiently by the fact that one almost never hears the full name used. For my money, we need a name which is short enough for everyone to use, yet which sufficiently denominates what we're talking about (so "Open API", which I do frequently hear, is not good enough.)

I understand your reservations about the use of "Mojaloop API", and I would also support a name which would make the purpose of the API immediately clear. Would an alternative name meet your objections?

**Comment by ehenrka** [comment]
The change request was to change the name to Mojaloop API, I'm just answering that I'm not in favor of changing the name to Mojaloop API.

Another name might be fine, preferably something which includes terms "FSP", "Interoperability" and "API". Maybe just the simpler "FSP Interoperability API", which does not restrict usage to Mojaloop switch, it explains what the API does, and would be simple to change both documentation wise (simple text change) and headers (FSPIOP is still valid).

**Comment by elnyry** [comment]
Thanks for the feedback @ehenrka ..

**Comment by elnyry** [comment]
Thanks for the feedback @ehenrka ..

The point you made about _rebranding cost within platforms_ is one that I agree with - but there is also the question of whether or not making those changes for all the platforms is mandatory or even necessary..

One point I have to disagree with is - "_Changing the name to Mojaloop more or less restricts the API to be for the Mojaloop switch instead of being open, as the name Mojaloop is tightly coupled with the open source switch implementation._" I think the API stands on its own and is not tightly coupled in anyway with the Open Source Switch implementation, though its true that the Implementation is tied to the API. The other reasons may be more convincing but I do not see why making this name change restricts the API, in reference to this point. And, the Mojaloop API is Open isn't it, where is it restricted as opposed to an "Open API"?

As you know, 'moja-loop' meaning 'one-loop' does in fact, convey a meaning :-) about interoperability between platforms, I guess for those who want to look for it. I've never heard anyone using the FSPIOP name outside of the CCB and the Mojaloop Core team. I've witnessed hours spent on fixing bugs because of spelling errors in code, where a lot of times "FSIOP" or "FSPOIP" were used instead of the "FSPIOP" term.. So this does make it simpler.

I also like retaining the header names within the API itself, to actually leave that legacy which I'm sure will be appreciated by those who are closely involved with the API (whatever it is going to be called :-) )

**Comment by ehenrka** [comment]
We would need to update our product documentation that is used by our customers. We can't say in our documentation that we support an API that have changed it's name, as we can't just expect everyone to know about the name change. Some readers would probably also just think that we support integrations to the Mojaloop switch based on the name. This means that we would have to explain that the API is not tightly coupled to the Mojaloop switch and could be used for other purposes like a simple bilateral setup or other switches not based on Mojaloop.

If we would change the name to Mojaloop API, then that will be tightly coupled by its name to the Mojaloop switch. Sorry for repeating myself, but if you were to startup a new switch implementation that is meant to "compete" with Mojaloop but not based on Mojaloop (the open source switch implementation), would you rather use an API named Mojaloop API instead of one named something which is not using another existing competing switch name? This limits competition in the market.

Mojaloop might mean something for those who have heard the explanation, unless you know both swahili and english, but I would guess that there are more people that would understand the word interoperability. If someone can't spell or copy-paste the header name FSPIOP-[name], then why should it be easier to spell or copy-paste the name Mojaloop-[name]?

**Comment by MichaelJBRichards** [comment]
... or maybe FSP InterApi...?

**Comment by matthewbohan** [comment]
Good discussion.  I hear the concern and understand that, as an example, the folks at Microsoft Windows might be reluctant to implement a "Linux API."  That said, in Pakistan, where a non-Mojaloop based solution is being implemented, I don't think the Mojaloop name would be a barrier to implementing this API to connect DFSPs to that switch.  With respect to additional adoptions outside of Mojaloop, are we seeing any?  We have a couple of implementations of the Mojaloop hub going on that will use the API, and we have already seen those implementers joining our current governance group.   We are expecting that group to continue growth, but it would be great to also add folks who are non-Mojaloop users to the group if they are out there.  Are any of our CCB members hearing of interest, or are any of your organizations evangelizing for this standard with other switch providers?  If so, are they interested?  Can we ask them if the Mojaloop name would be a barrier to their adoption or usage?  Would the Mojaloop name or any other name change the level of evangelizing that your organizations do for the standard?

I also hear the concern that it will not be free to implement a name change, so if we do make a change, it should get us something.  I'll also add the counterpoint that it will never be less expensive to implement a name change than it is right now.

Michael's point that you never hear someone use the full name is an important one.  Branding is more important to adoption than most of us give it credit for.  We will continue to put a shoulder behind the Mojaloop branding, and would like the API to be included in that messaging.  The Mojaloop brand is intended to be about inclusiveness, to create "one loop," and help things to work together.  Perhaps the brand name is falling short of its aspirations, though I haven't heard any clear evidence of that.  Perhaps adding the API into that branding would make those intentions more clear?  I can tell you that it is much easier to make an impression about the API using the Mojaloop name and in combination with messaging about the Mojaloop hub.  I have personally tried both with multiple audiences.

There is also some foreshadowing for a related discussion around governance.  We have talked about the formation of an entity to own and govern Mojaloop outside of the Gates foundation, and we have further talked about that organization as sponsor for this governance group, which we have been calling the change control board.  Perhaps who governs a standard is less visible than the name of the standard,  and to revert to a previous example, Microsoft has gotten involved with Linux contributions and, to some degree, its governance.  In fairness, I realize that took a while.  If the group has concerns about this potential direction, then we should start to discuss alternatives.

In summary, I think there is clear benefit to a shorter, more memorable name with respect to adoption and usage.  I know that the foundation will continue to invest in market impressions around the Mojaloop name.  I believe that Mojaloop adopters like Mowali will invest to some degree in market impressions around the Mojaloop name.  I would be hesitant to forego those benefits based on the conjecture that some future competitor (who otherwise would adopt the API) may not adopt the API based on the Mojaloop name.

**Comment by ehenrka** [comment]
@matthewbohan I would think that Mojaloop (as in the open source switch) also defines and supports other APIs than "The Open API for FSP Interoperability"? For example, there have been discussions regarding a settlement interface, and I would also assume that there are ways to control Mojaloop using an API (some front-end API). For me, the most correct way would then be to use Mojaloop as the API suite/set name or similar (e.g. the "Mojaloop API Suite"), which then includes a set of APIs, where "The Open API for FSP Interoperability" (or just simpler "FSP Interoperability API") would be one of the APIs, and there could also be a settlement API and a front-end API named something else in the set.

This would allow us to use Mojaloop APIs in branding without mentioning the exact API name in for example high-level presentations, but also continue to internally and in detailed developer presentations use the more technically correct name of this API, and to be able separate the different APIs that Mojaloop actually supports (e.g. the mentioned settlement interface). Would this be an acceptable way forward for you?

**Comment by matthewbohan** [comment]
@ehenrka I think that approach makes a lot of sense, and will discuss with folks here and a couple of others.  Any comments from others on this thread?  Any discussion on the governance issue mentioned?

**Comment by elnyry** [comment]
@matthewbohan @ehenrka I do like the "Mojaloop API Suite" approach as we currently have two other APIs in addition to the FSPIOP API - an API for Settlements and Hub Operations, so all these can be part of the Suite.

**Comment by MichaelJBRichards** [comment]
Yes, makes sense to me; and I like @ehenrka's suggestion for renaming the Open API too.

**Comment by ritviksinha** [comment]
@matthewbohan @ehenrka I agree. It makes a lot of sense to have Mojaloop API Suite with APIs grouped together as FSP Interoperability API, Settlements API etc.

**Comment by rjpwip** [comment]
@matthewbohan @ehenrka 
Nice compromise re: "Mojaloop API Suite".
+1 from me

**Comment by jmssebunnya** [comment]
*Mojaloop API Suite* Bundles it together in a good way.

Best Regards
Mark

**Comment by Mutashobya** [comment]
I am in support of the Mojaloop API Suite to bundle the current three APIs. Also supporting the rename of "The Open API for FSP Interoperability" to a much more shorter name.

**Comment by ehenrka** [comment]
This CR has been open for quite some time now without any action. To be able to close it as part of the next version (v1.2 or v2.0), I suggest to do at least the following:

1. Change all names in the document from "Open API for FSP Interoperability" to "FSP Interoperability API"
2. Add generic information about Mojaloop and that the API is part of the Mojaloop API Suite, preferably as part of Section 2 (Introduction). As part of this, also explain that the API can be used without Mojaloop as it could be another Switch or without a switch at all.

@matthewbohan Is there some styling in Word for Mojaloop that should be used so that the document styling is consistent with other Mojaloop documents?

**Comment by elnyry-sam-k** [comment]
Thanks Henrik (@ehenrka ). 

we have a style guide here: https://github.com/mojaloop/mojaloop-specification/tree/master/ccb-common-artifacts

**Comment by ehenrka** [comment]
Thanks @elnyry-sam-k, but that seems to be the one that we are using for the FSP Interoperability API today, I was thinking if there is a Mojaloop specific style that we should be using instead?

**Comment by elnyry-sam-k** [comment]
Good point Henrik; but thats the most recent style guide we have :-).. Since we now moved to markdown now, I don't think we have any such template. I'll bring this up during our next CCB call.

**Comment by elnyry-sam-k** [comment]
Closing this.

We settled with: "Mojaloop Family of APIs" for the while set and FSP Interoperability (FSPIOP) API
